Fix aspect-ratio items with percentage max-width collapsing to zero

yulunwu2 opened the pull request:
#### f7a85fb9452f0a72d1afc8a9564eca2b9c50722c
<pre>
Fix aspect-ratio items with percentage max-width collapsing to zero
<a href="https://bugs.webkit.org/show_bug.cgi?id=293914">https://bugs.webkit.org/show_bug.cgi?id=293914</a>
&lt;<a href="https://rdar.apple.com/152456882">rdar://152456882</a>&gt;

Reviewed by NOBODY (OOPS!).

During intrinsic sizing (grid track sizing or min-content containers),
containingBlockLogicalWidthForContent() may return 0 because the container&apos;s
width depends on the size of its children. When computeLogicalWidthFromAspectRatio()
resolved a percentage max-width against this zero available width, it
collapsed the intrinsic contribution to zero.

This PR updates RenderBlock::computePreferredLogicalWidths() to skip
min/max constraints during intrinsic sizing per spec:

<a href="https://www.w3.org/TR/css-sizing-3/#cyclic-percentage-contribution">https://www.w3.org/TR/css-sizing-3/#cyclic-percentage-contribution</a>:

&quot;If the box is non-replaced, then the entire value of any max size property
or preferred size property specified as an expression containing a percentage
that is cyclic is treated for the purpose of calculating the box&apos;s intrinsic
size contributions only as that property&apos;s initial value.&quot;
&quot;For the min size properties, a cyclic percentage is resolved against zero
for determining intrinsic size contributions.&quot;

* LayoutTests/fast/css/aspect-ratio-percent-max-width-expected.html: Added.
* LayoutTests/fast/css/aspect-ratio-percent-max-width.html: Added.
* LayoutTests/fast/grid/grid-item-aspect-ratio-expected.html: Added.
* LayoutTests/fast/grid/grid-item-aspect-ratio.html: Added.
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::computePreferredLogicalWidths):
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::computeIntrinsicKeywordLogicalWidths const):
(WebCore::RenderBox::computeUnconstrainedLogicalWidthFromAspectRatio const):
(WebCore::RenderBox::computeLogicalWidthFromAspectRatio const):
(WebCore::RenderBox::computeLogicalWidthFromAspectRatioInternal const): Deleted.
* Source/WebCore/rendering/RenderBox.h:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f7a85fb9452f0a72d1afc8a9564eca2b9c50722c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/161234 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/34707 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/27808 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/170137 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/117605 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/ca8d2592-4cf5-4c72-a7ee-447f105f85dc) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/163103 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/34808 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/34708 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/125072 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/88157 "2 flakes 1 failures") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/c24d7bd1-0761-42cb-becb-3e2789cbf433) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/164193 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/27359 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/144839 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/105669 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/65ce7c3a-8645-4ada-a7b7-aa713b94a3a9) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/26407 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/24909 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/17857 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/136101 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/22599 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/172620 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/19641 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/24240 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/133348 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/34381 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/29002 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/133357 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/34366 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/144394 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/96231 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/27908 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/21212 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/33876 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/101301 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/33375 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/33619 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/33524 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/11e53463253da491701f5893a369052815d7c461)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118029 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37707 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28343 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124179 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70065 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c2bf8961-9deb-4f9f-96ea-9db62969255d) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119907 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38400 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46289 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89568 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59168 "An unexpected error occured. Recent messages:Printed configuration") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/1caa75f6-e9cc-4e6d-8d10-e796ea5b76a3) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/120981 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30584 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105833 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70060 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/1c9058bf-6c8b-40ca-beaa-13b1acef8cf4) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29652 "Passed tests") | [⏳ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/macOS-Tahoe-Debug-API-Tests-EWS "Waiting to run tests") | [⏳ 🛠 gtk3-libwebrtc ](https://ews-build.webkit.org/#/builders/GTK-GTK3-LibWebRTC-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100007 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23950 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127260 "Built successfully") | | 
| [⏳ 🛠 🧪 jsc-debug-arm64 ](https://ews-build.webkit.org/#/builders/JSC-Tests-O3-Debug-arm64-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios-safer-cpp ](https://ews-build.webkit.org/#/builders/Apple-iOS-26-Safer-CPP-Checks-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24128 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98241 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44932 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33856 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98027 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45293 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102057 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41367 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43424 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21412 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44804 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50478 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44264 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47609 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45953 "Built successfully") | | | | </details>

yulunwu2 commented on LayoutTests/fast/grid/grid-item-aspect-ratio.html:
I'll be sure to migrate the test over in a follow up PR :)

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/b5938becc109b41331b17c91926906e4c135aa4f)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/157172 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/30509 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/23700 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/165995 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/111254 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/a5166f8e-dcd5-44e7-bab3-ec65faa6f17d) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/159043 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/30644 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/30511 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/121717 "Found 1 new test failure: fast/grid/grid-item-aspect-ratio.html (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/85462 "3 flakes 5 failures") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/52421180-ae94-4c1f-8112-653b1890ce3b) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/160130 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/23952 "Found 1 new test failure: fast/grid/grid-item-aspect-ratio.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/141130 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/102385 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/17d9db8a-6b63-4285-9cfe-306fe03ab89f) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/23007 "Exiting early after 10 failures. 10 tests run. 1 failures") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/21256 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/13767 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/132687 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/18958 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/168480 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/12639 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/20578 "Found 1 new test failure: fast/grid/grid-item-aspect-ratio.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/129849 "Found 1 new test failure: fast/grid/grid-item-aspect-ratio.html (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/30110 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/25334 "Found 1 new test failure: fast/grid/grid-item-aspect-ratio.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/129957 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/30033 "Built successfully") | [❌ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/140752 "Exiting early after 10 failures. 10 tests run. 1 failures") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/87854 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/24778 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/17556 "Found 1 new test failure: fast/grid/grid-item-aspect-ratio.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/29744 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/93758 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/29266 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/29496 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/29393 "Built successfully") | | | | </details>

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/8d70084429f5d9f592defb640bdd6b9bedc96ee2)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/157885 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/31222 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/24415 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/166713 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/111968 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/159756 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/31357 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/31224 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/122263 "Found 117 new test failures: css2.1/20110323/border-collapse-offset-002.htm fast/css/bidi-override-in-anonymous-block.html fast/css/caption-width-relative-position-offset-top.htm fast/css/caption-width-relative-position.htm fast/dom/HTMLTableElement/colSpan.html fast/dom/HTMLTableElement/createCaption.html fast/repaint/table-section-repaint.html fast/table/frame-and-rules.html imported/w3c/web-platform-tests/css/CSS2/tables/border-collapse-offset-002.xht imported/w3c/web-platform-tests/css/css-break/table/table-collapsed-borders-paint-htb-ltr.html ... (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/85845 "1 flakes 4 failures") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/160843 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/24552 "Found 60 new test failures: css2.1/20110323/border-collapse-offset-002.htm fast/css/bidi-override-in-anonymous-block.html fast/css/caption-width-relative-position-offset-top.htm fast/css/caption-width-relative-position.htm fast/dom/HTMLTableElement/colSpan.html fast/dom/HTMLTableElement/createCaption.html fast/forms/ios/drag-range-track.html fast/table/frame-and-rules.html tables/mozilla/bugs/bug1163-1.html tables/mozilla/bugs/bug1302.html ... (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/141789 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/102929 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/23608 "Found 22 new test failures: imported/w3c/web-platform-tests/css/CSS2/tables/border-collapse-offset-002.xht imported/w3c/web-platform-tests/css/css-break/table/table-collapsed-borders-paint-htb-ltr.html imported/w3c/web-platform-tests/css/css-break/table/table-collapsed-borders-paint-vlr-rtl.html imported/w3c/web-platform-tests/css/css-break/table/table-collapsed-borders-paint-vrl-ltr.html imported/w3c/web-platform-tests/css/css-break/table/table-grid-paint-htb-ltr.html imported/w3c/web-platform-tests/css/css-break/table/table-grid-paint-vlr-rtl.html imported/w3c/web-platform-tests/css/css-break/table/table-grid-paint-vrl-rtl.html imported/w3c/web-platform-tests/css/css-flexbox/table-as-item-inflexible-in-column-2.html imported/w3c/web-platform-tests/css/css-flexbox/table-as-item-inflexible-in-row-2.html imported/w3c/web-platform-tests/css/css-grid/grid-items/grid-items-percentage-margins-002.html ... (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/21902 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/14486 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/133289 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/19594 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/169203 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/14058 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/21217 "Found 60 new test failures: css2.1/20110323/border-collapse-offset-002.htm fast/css/bidi-override-in-anonymous-block.html fast/css/caption-width-relative-position-offset-top.htm fast/css/caption-width-relative-position.htm fast/dom/HTMLTableElement/colSpan.html fast/dom/HTMLTableElement/createCaption.html fast/forms/placeholder-content-center.html fast/repaint/table-section-repaint.html fast/shapes/shape-outside-floats/shape-outside-floats-border-radius-padding-box-000.html fast/table/frame-and-rules.html ... (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/130436 "Found 116 new test failures: css2.1/20110323/border-collapse-offset-002.htm fast/css/bidi-override-in-anonymous-block.html fast/css/caption-width-relative-position-offset-top.htm fast/css/caption-width-relative-position.htm fast/dom/HTMLTableElement/colSpan.html fast/dom/HTMLTableElement/createCaption.html fast/repaint/table-section-repaint.html fast/table/frame-and-rules.html imported/w3c/web-platform-tests/css/css-break/table/table-collapsed-borders-paint-htb-ltr.html imported/w3c/web-platform-tests/css/css-break/table/table-collapsed-borders-paint-vlr-rtl.html ... (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/30968 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/25969 "Found 60 new test failures: accessibility/mac/input-replacevalue-userinfo.html css2.1/20110323/border-collapse-offset-002.htm fast/css/bidi-override-in-anonymous-block.html fast/css/caption-width-relative-position-offset-top.htm fast/css/caption-width-relative-position.htm fast/dom/HTMLTableElement/colSpan.html fast/dom/HTMLTableElement/createCaption.html fast/parser/entities-in-html.html fast/repaint/table-section-repaint.html fast/table/frame-and-rules.html ... (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/130550 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/30906 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/141388 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/88759 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/25287 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/18194 "Exiting early after 60 failures. 78565 tests run. 60 failures") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/30458 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/95471 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/29979 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/30209 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/30106 "Built successfully") | | | | </details>

yulunwu2 committed:
Don't return 0 for indefinite grid area widths in containingBlockLogicalWidthForContent
https://bugs.webkit.org/show_bug.cgi?id=293914
<rdar://152456882>

Reviewed by NOBODY (OOPS!).

containingBlockLogicalWidthForContent() incorrectly collapsed
indefinite widths to zero as a side effect of "value_or(0)".

This PR updates to only return a width of 0 for definite width
items with a size of zero and fall through to

containingBlock->contentBoxLogicalWidth()

for indefinite items.

* LayoutTests/fast/grid/grid-item-aspect-ratio-expected.html: Added.
* LayoutTests/fast/grid/grid-item-aspect-ratio.html: Added.
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::containingBlockLogicalWidthForContent const):
(WebCore::RenderBox::computePreferredLogicalWidths):
* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::RenderGrid::placeItemsOnGrid):

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/405727452a516d64e4209c4be78d53662a78bb11)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/158033 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/31370 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/24563 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/166861 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/112116 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/159904 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/31507 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/31372 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/122387 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/85918 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/160991 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/24676 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/141932 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/103055 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/23732 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/22038 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/14634 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/133416 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/19726 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/169351 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/14705 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/21349 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/130561 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/31116 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/26103 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/130676 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/31054 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/141520 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/88950 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/25409 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/18326 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/30606 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/96139 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/30127 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/30357 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/30254 "Built successfully") | | | | </details>

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/5aa43d8463dde8fde8e708db0c634ae31df10fc9)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/159157 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/32585 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/25690 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/167986 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/113241 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/161026 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/32653 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/32572 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/123300 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/86572 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/162114 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/25574 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/142963 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/103966 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/24628 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/23050 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/15759 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/134314 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/20743 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/170481 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/16221 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/22369 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/131495 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/32274 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/27119 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/131607 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/32218 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/142536 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/90271 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/26308 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/19345 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/31729 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/97743 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/31249 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/31522 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/31404 "Built successfully") | | | | </details>

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/fe1e6d4abf811c5350fc243e00053072144fed94)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/159408 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/32836 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/25944 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/168240 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/113786 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/161277 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/32904 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/32824 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/123507 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/86692 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/162365 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/25752 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/143179 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/104171 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/24824 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/23261 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/16011 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/134492 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/20959 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/170732 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/16766 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/22585 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/131712 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/32525 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/27336 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/131825 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/32469 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/142752 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/90594 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/26483 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/19561 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/31980 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/98432 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/31500 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/31773 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/31655 "Built successfully") | | | | </details>

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/c4dbb233086bd8a326101fac99da1a76d4753436)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/159507 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/32974 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/26052 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/168345 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/113888 "Built successfully") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/ba8c9e62-55e6-4d68-b854-544b09139e2f) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/161376 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/33008 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/32927 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/123594 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/86858 "8 failures") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/94bb0491-a5c1-48ae-93cb-36e36ef99d9e) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/162464 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/25850 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/143273 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/104252 "Passed tests") | | [❌ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/cacca285-c004-45d7-9ea8-c4e080645753) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/24903 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/23356 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/16115 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/134594 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/21045 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/170838 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/16875 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/22676 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/131812 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/32629 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/27432 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/131909 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/32638 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/142838 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/90720 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/26559 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/19652 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/32147 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/98543 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/31645 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/31878 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/31795 "Built successfully") | | | | </details>

webkit-early-warning-system commented:
EWS run on previous version of this PR (hash https://github.com/WebKit/WebKit/commit/1ab3eb7433bc533753adfd9f9296832ad1c30ab5)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/160118 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/33588 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/26692 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/169010 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/114499 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/e9e00670-4832-4fc3-991d-8e81197e172b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/161987 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/33691 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/33590 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/124113 "Found 2 new test failures: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-022.html imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-024.html (failure)") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/87048 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/97a8d09e-0f8e-4678-ad22-5594881b0b9c) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/163076 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/26361 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/143821 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/104722 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/7fc67b81-6fa4-44e9-89c3-99de42bbb5db) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/25414 "Found 2 new test failures: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-022.html imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-024.html (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/23906 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/16739 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/135106 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/21584 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/171488 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/17486 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/23223 "Found 2 new test failures: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-022.html imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-024.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/132372 "Found 2 new test failures: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-022.html imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-024.html (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/33264 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/27993 "Found 2 new test failures: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-022.html imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-024.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/132398 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/33249 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/143379 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/91449 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/27013 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/20192 "Found 2 new test failures: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-022.html imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/grid-aspect-ratio-024.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/32758 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/99155 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/32256 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/32502 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/32406 "Built successfully") | | | | </details>

webkit-early-warning-system commented:
EWS run on current version of this PR (hash https://github.com/WebKit/WebKit/commit/5f9a169600fede7a39923df91ea121de65083052)<details>

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/160131 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/33601 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/26705 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/169025 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/114512 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/2c9bcf32-7a07-4f1a-aede-16e5ccc9c094) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/162000 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/33704 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/33603 "Built successfully") | [❌ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/124126 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/87060 "2 failures") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/75c9570d-d2e7-4e20-8344-2554f3f8d499) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/163089 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/26373 "Passed tests") | [❌ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/143835 "Found 1 new API test failure: TestWebKitAPI.WKWebExtensionAPIScripting.ExecuteScriptWithUserGesture (failure)") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/104735 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/7034611f-7cb9-4d37-befa-0098fa8f0efd) 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/25426 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/23921 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/16752 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/135118 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/21598 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/171502 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/17499 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/23238 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/132386 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/33277 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/28007 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/132412 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/33262 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/143393 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/91482 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/27024 "Passed tests") | [❌ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/20207 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-sizing/aspect-ratio/block-aspect-ratio-056.html (failure)") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/32771 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/99168 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/32269 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/32515 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/32419 "Built successfully") | | | | </details>

yulunwu2 committed:
Fix aspect-ratio items with percentage max-width collapsing to zero
https://bugs.webkit.org/show_bug.cgi?id=293914
<rdar://152456882>

Reviewed by NOBODY (OOPS!).

This PR fixes an issue where items with aspect ratio and percentage-based
max-width were collapsing to zero width during intrinsic sizing, affecting
items inside grid tracks, flex containers, etc by implementing two parts of
the CSS spec.

This PR updates RenderBox::computeLogicalWidthFromAspectRatio() to skip
min/max constraints during intrinsic sizing.

https://www.w3.org/TR/css-sizing-3/#cyclic-percentage-contribution):

"If the box is non-replaced, then the entire value of any max size property
or preferred size property specified as an expression containing a percentage
that is cyclic is treated for the purpose of calculating the box's intrinsic
size contributions only as that property's initial value."
"For the min size properties, a cyclic percentage is resolved against zero
for determining intrinsic size contributions."

This PR also implements transferred size suggestion for grid item widths per:

https://drafts.csswg.org/css-grid-2/#transferred-size-suggestion

The content-based minimum size for a grid item in a given dimension is its
specified size suggestion if it exists, otherwise its transferred size suggestion
if that exists and the element is replaced, else its content size suggestion.

If the item has a preferred aspect ratio and its preferred size in the opposite
axis is definite, then the transferred size suggestion is that size (clamped
by the opposite-axis minimum and maximum sizes if they are definite),
converted through the aspect ratio. It is otherwise undefined.

* LayoutTests/fast/css/aspect-ratio-percent-max-width-expected.html: Added.
* LayoutTests/fast/css/aspect-ratio-percent-max-width.html: Added.
* LayoutTests/fast/grid/grid-item-aspect-ratio-expected.html: Added.
* LayoutTests/fast/grid/grid-item-aspect-ratio.html: Added.
* Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp:
(WebCore::transferredSizeFromAspectRatio):
(WebCore::GridTrackSizingAlgorithmStrategy::minContributionForGridItem const):
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::computePreferredLogicalWidths):
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::computeIntrinsicKeywordLogicalWidths const):
(WebCore::RenderBox::computeUnconstrainedLogicalWidthFromAspectRatio const):
(WebCore::RenderBox::computeUnconstrainedLogicalHeightFromAspectRatio const):
(WebCore::RenderBox::computeLogicalWidthFromAspectRatio const):
(WebCore::RenderBox::computeLogicalWidthFromAspectRatioInternal const): Deleted.
* Source/WebCore/rendering/RenderBox.h:
(WebCore::RenderBox::isUnresolveableStretchSize const):

sammygill commented on Source/WebCore/rendering/RenderBox.cpp:
```diff
@@ -5231,9 +5231,23 @@ LayoutUnit RenderBox::computeLogicalWidthFromAspectRatioInternal() const
+    // preferred size property specified as an expression containing a percentage that is
+    // cyclic is treated for the purpose of calculating the box's intrinsic size contributions
+    // only as that property's initial value."
+    if (needsPreferredLogicalWidthsUpdate()
```
Why is the preferred widths bits being dirty the indicator that there is a cyclic dependency here? That does not seem right since those bits don't have anything to do with that and instead we would probably need some other mechanism to take care of this imo

yulunwu2 commented on Source/WebCore/rendering/GridTrackSizingAlgorithm.cpp:
```diff
@@ -1038,6 +1038,29 @@ LayoutUnit GridTrackSizingAlgorithmStrategy::logicalHeightForGridItem(RenderBox&
+// opposite axis is definite, the transferred size suggestion is that size
+// (clamped by the opposite-axis minimum and maximum sizes if they are definite),
+// converted through the aspect ratio. It is otherwise undefined.
+static std::optional<LayoutUnit> transferredSizeFromAspectRatio(const RenderBox& gridItem, bool isInlineAxis)
```
Removing transferredSizeFromAspectRatio from this PR

alanbaradlay commented on Source/WebCore/rendering/RenderBox.cpp:
```diff
@@ -5239,9 +5232,17 @@ LayoutUnit RenderBox::computeLogicalWidthFromAspectRatioInternal() const
+    // only as that property's initial value."
+    if (computingIntrinsicWidth == ComputingIntrinsicWidth::Yes
+        && (style().logicalMaxWidth().isPercentOrCalculated() || style().logicalMinWidth().isPercentOrCalculated()))
+        return logicalWidth;
```
I also wonder why this logic needs to live in a function that's called by both layout and min/max width computation (while we have dedicated functions like `RenderBox::computePreferredLogicalWidths` where we already deal with transferred size computation for min/max)

sammygill replied:
Isn't the lack of infrastructure the underlying issue? If the problem that causes the bug is that we cannot determine whether we are in the cyclic case then we should figure out the best way to solve that problem directly. If that means we need to figure out a way to have more information here then that is what we should do or maybe there is another way to solve the problem.

With respect to the current approach, I don't think we should coerce the value to be something specific in certain cases but wrong in other cases and hope that it gets fixed up later. This code is specifically about computing the preferred widths so we should make sure that we compute the right value in all cases since we have no control over what the caller will do afterwards. If we had a % that was resolvable with respect to computing the preferred widths of the renderer but that is not what we ended up storing and returning then that would probably be wrong

yulunwu2 committed:
Fix aspect-ratio items with percentage max-width collapsing to zero
https://bugs.webkit.org/show_bug.cgi?id=293914
<rdar://152456882>

Reviewed by NOBODY (OOPS!).

During intrinsic sizing (grid track sizing or min-content containers),
containingBlockLogicalWidthForContent() may return 0 because the container's
width depends on the size of its children. When computeLogicalWidthFromAspectRatio()
resolved a percentage max-width against this zero available width, it
collapsed the intrinsic contribution to zero.

This PR updates RenderBlock::computePreferredLogicalWidths() to skip
min/max constraints during intrinsic sizing per spec:

https://www.w3.org/TR/css-sizing-3/#cyclic-percentage-contribution:

"If the box is non-replaced, then the entire value of any max size property
or preferred size property specified as an expression containing a percentage
that is cyclic is treated for the purpose of calculating the box's intrinsic
size contributions only as that property's initial value."
"For the min size properties, a cyclic percentage is resolved against zero
for determining intrinsic size contributions."

* LayoutTests/fast/css/aspect-ratio-percent-max-width-expected.html: Added.
* LayoutTests/fast/css/aspect-ratio-percent-max-width.html: Added.
* LayoutTests/fast/grid/grid-item-aspect-ratio-expected.html: Added.
* LayoutTests/fast/grid/grid-item-aspect-ratio.html: Added.
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::computePreferredLogicalWidths):
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::RenderBox::computeIntrinsicKeywordLogicalWidths const):
(WebCore::RenderBox::computeUnconstrainedLogicalWidthFromAspectRatio const):
(WebCore::RenderBox::computeLogicalWidthFromAspectRatio const):
(WebCore::RenderBox::computeLogicalWidthFromAspectRatioInternal const): Deleted.
* Source/WebCore/rendering/RenderBox.h:
